### DISCUSSION TITLE
Get all tests to wait at a certain point in a workflow and then execute all at once after ramp up finishes

### DISCUSSION THREAD

**Discussion by ghost** [artilleryio/artillery#954] [Artillery]
Hello, fellow performance testers,

I'm new to the board and have been stuck on a scenario I'm trying to put together and was curious if anyone got some insight or has done something similar before.

So the problem I'm facing is that I have a website that I want to have a vast amount of users login find an item they desire(RNG) and then wait till everyone else finds an item they like and get all the tests to simultaneously hit a save button.

Like my first thought was to use a `think` but with that, it's not going to perform the way I want it to because if I have a batch of tests that start and get to a point before the save button, while the other tests are still ramping up the tests at the beginning might start hitting the save button which defeats the purpose of what I'm trying to test.

Has anyone dealt with a similar situation in the past, much appreciated for any feedback.

**Comment by hassy** [comment]
hey @Right10Privacy10👋

VUs in Artillery are independent of each other (by design, to mimic API clients and browsers in the real world). If you want to coordinate between them, you will need to add dependency on an external store of some sort, e.g. Redis.

With Redis (for example) you could have VUs subscribe to a pubsub channel, pause halfway through a scenario and wait for a message on that channel before continuing with their scenario.

**Reply by ghost** [reply]
ahh okay, so then it would be something like creating a function that will be executed within the scenario for every VU that will wait for an external source(me) to tell the test to continue?

**Reply by ghost** [reply]
Nvm silly question on my part, managed to figure it out, thank you for the hint !!!
